### DISCUSSION TITLE
Script email with attachments and save to Drafts?

### DISCUSSION THREAD

**Discussion by jamesaoverton** [pimalaya/himalaya#258] [General]
Thank you for this lovely tool. I was able to get Himalaya working without trouble, but there's one thing I want to do that I can't figure out:

I want to write a shell script that creates an email and saves it to my Drafts folder with an attached file.

I think that none of the existing commands do quite what I want:

1. `himalaya template new --subject ... > draft.mail` will create a local file with the content I want, but does not add attachments or save to Drafts
2. `himalaya --mailbox Drafts save < draft.mail` saves to Drafts, but does not add attachments
3. `himalaya write --attachment ...` creates an email with attachments and can save to Drafts, but it's interactive not scripted

I also tried `himalaya write < draft.mail` which seemed promising, but I got an endless stream of `(s)end, (e)dit, (l)ocal/(r)emote draft or (d)iscard? empty choice` lines until I pressed Ctrl-C.

Is there some combination of Himalaya commands that will let me script a draft email with attachments?

Thanks again!

**Comment by soywod** [comment]
> Thank you for this lovely tool. I was able to get Himalaya working without trouble

Happy to hear that :) big features are coming, stay tuned!

> I want to write a shell script that creates an email and saves it to my Drafts folder with an attached file.

I'm afraid it is not possible right now, it is a special use case I did not think about. There is no way for the command `send` or `save` to add an attachment. I will switch this discussion to an issue, thanks for your feedback!

**Comment by soywod** [comment]
It is doable now thanks to the MML language introduced a while ago. See the [FAQ](https://github.com/pimalaya/himalaya?tab=readme-ov-file#faq): "How to compose a message?".
